src/storage/mod.rs

```diff
@@ -1930,4 +2123,292 @@ mod tests {
+            .collect();
+        expect_multi_values(
+            results,
+            storage.async_raw_batch_get(Context::new(), keys).wait(),
+        );
+
+        // Delete ["b", "d"]
+        storage
+            .async_raw_batch_delete(
+                Context::new(),
+                vec![b"b".to_vec(), b"d".to_vec()],
+                expect_ok_callback(tx.clone(), 1),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        // Assert "b" and "d" are gone
+        expect_value(
+            b"aa".to_vec(),
+            storage.async_raw_get(Context::new(), b"a".to_vec()).wait(),
+        );
+        expect_none(storage.async_raw_get(Context::new(), b"b".to_vec()).wait());
+        expect_value(
+            b"cc".to_vec(),
+            storage.async_raw_get(Context::new(), b"c".to_vec()).wait(),
+        );
+        expect_none(storage.async_raw_get(Context::new(), b"d".to_vec()).wait());
+        expect_value(
+            b"ee".to_vec(),
+            storage.async_raw_get(Context::new(), b"e".to_vec()).wait(),
+        );
+
+        // Delete ["a", "c", "e"]
+        storage
+            .async_raw_batch_delete(
+                Context::new(),
+                vec![b"a".to_vec(), b"c".to_vec(), b"e".to_vec()],
+                expect_ok_callback(tx.clone(), 2),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        // Assert no key remains
+        for (k, _) in test_data {
+            expect_none(storage.async_raw_get(Context::new(), k).wait());
+        }
+    }
+
+    #[test]
+    fn test_raw_scan() {
+        let read_pool = new_read_pool();
+        let config = Config::default();
+        let mut storage = Storage::new(&config, read_pool).unwrap();
+        storage.start(&config).unwrap();
+        let (tx, rx) = channel();
+
+        let test_data = vec![
+            (b"a".to_vec(), b"aa".to_vec()),
+            (b"a1".to_vec(), b"aa11".to_vec()),
+            (b"a2".to_vec(), b"aa22".to_vec()),
+            (b"b".to_vec(), b"bb".to_vec()),
+            (b"ba".to_vec(), b"bbaa".to_vec()),
+            (b"bb".to_vec(), b"bbbb".to_vec()),
+            (b"bc".to_vec(), b"bbcc".to_vec()),
+            (b"c".to_vec(), b"cc".to_vec()),
+            (b"d".to_vec(), b"dd".to_vec()),
+            (b"e".to_vec(), b"ee".to_vec()),
+        ];
+
+        // Write key-value pairs in batch
+        storage
+            .async_raw_batch_put(
+                Context::new(),
+                test_data.clone(),
+                expect_ok_callback(tx.clone(), 0),
+            )
+            .unwrap();
+        rx.recv().unwrap();
+
+        // Scan pairs with key only
+        let results = test_data
+            .iter()
+            .map(|&(ref k, _)| Some((k.clone(), vec![])))
+            .collect();
+        expect_multi_values(
+            results,
+            storage
+                .async_raw_scan(Context::new(), vec![], 10, true)
```

Comment:
The test should also cover the case that `start_key` is not empty.

Reply:
Updated test cases to cover these cases